library/src/scala/Tuple.scala

```diff
@@ -199,7 +199,7 @@ object Tuple {
   def apply(): EmptyTuple = EmptyTuple
 
   /** Tuple with one element */
-  def apply[T](x: T): T *: EmptyTuple = Tuple1(x)
+  def apply[T](x: T): Tuple1[T] = Tuple1(x)
```

Comment:
The root of the issue seems to be that pattern matcher looks at `TupleN` but not `*:`. We should probably fix that instead. We are not supposed to use `Tuple1` explicitly as in the distant this might just be an erasure detail.

Reply:
For pattern matcher, the specification requires `_N` to be present. It's not clear how to improve the specification to support `*:` without complicating things.

Actually, I think it might be a feature creep, I am not sure if it should be supported. The representation of tuples as `*:` is an implementation detail. The users stay with `(a, b, c, ...)`, which desugars to `TupleN`, which in turn extends `*:`.

Reply:
I think we can create an issue to discuss whether support `*:` and how to support it as `unapply` result type. WDYT?

Reply:
Yes. Because the idea of having generic tuples is to not rely on `TupleN` anymore.  At some point in the future we plan to remove `TupleN[X1, ...]` and only use `(X1, ...)` or `X1 *: ...`. Then we will only rely on the `unapply` with `*:`.

Reply:
Just created #11008.

Reply:
BTW, this PR also discovered some accidentally supported usage that should be invalid (see the changes):

- `tests/pos/i6199a.scala`
- `scala3doc/src/dotty/renderers/MemberRenderer.scala `